File: shanny-blog-java/blog-pojo/src/main/java/com/shanny/entity/TimeLine.java
package com.shanny.entity;

import com.shanny.converter.LongListConverter;
import jakarta.persistence.*;
import jakarta.validation.constraints.NotNull;
import jakarta.validation.constraints.Size;
import lombok.Data;
import org.springframework.format.annotation.DateTimeFormat;

import java.time.LocalDateTime;
import java.util.List;

@Data
@Entity
@Table(name = "timelines")
public class TimeLine {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @NotNull
    @Column(nullable = false)
    private Long articleId;

    @NotNull
    @DateTimeFormat(pattern = "yyyy-MM-dd HH:mm:ss")
    @Column(nullable = false)
    private LocalDateTime time;

    @NotNull
    @Column(length = 50, nullable = false)
    private String title;

    @NotNull
    @Column(length = 500, nullable = false)
    private String text;
}

File: shanny-blog-net/blog-pojo/Entities/TimeLine.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace blog_pojo.Entities
{
    [Table("timelines")]
    public class TimeLine
    {
        [Key]
        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        [Column("id")]
        public long Id { get; set; }

        [Required]
        [Column("articleId")]
        public long ArticleId { get; set; }

        [Required]
        [Column("time")]
        public DateTime Time { get; set; }

        [MaxLength(50)]
        [Required]
        [Column("title")]
        public string Title { get; set; } = string.Empty;

        [MaxLength(500)]
        [Required]
        [Column("text")]
        public string Text { get; set; } = string.Empty;
    }
}
